open-craft/sprints
# [BB-1620] Prepare API for billable/non-billable hours

This introduces the sustainability dashboard API. It returns the data in the following format:
```js
{
    "billable_accounts": [
        {
            "key": "ACCOUNT_KEY",
            "name": "ACCOUNT NAME",
            "overall": 10.0,
            "by_cell": {
                "Cell 1": 6.0,
                "Cell 2": 4.0
            },
            "by_person": {
                 "User 1": 1.0,
                 "User 2": 5.0,
                 "User 3": 4.0
            }
        },
    ]
    "non_billable_accounts": [
        …
    ],
    "non_billable_responsible_accounts": [
        …
    ]
}
```

## Testing instructions:
1. Set `.env` file.
1. Start the backend with `docker-compose -f local.yml up --build`.
1. Create superuser with `docker-compose -f local.yml run --rm django python manage.py createsuperuser`.
1. Log into the admin account `http://localhost:8000/admin/`.
1. Execute `http://localhost:8000/sustainability/dashboard/?from=2019-07-01&to=2019-08-01` in the browser. <s>Currently it takes about 40s, but I'm trying to make this async now.</s> After adding `multiprocessing` it takes around 14s.
1. Check if the values for your name look fine for these accounts.
